server/option.go

```diff
@@ -232,6 +232,14 @@ func WithRegistryInfo(info *registry.Info) Option {
 	}}
 }
 
+func WithRegistryInfoSkipListenAddr(skipListenAddr bool) Option {
```

Comment:
This option is not necessary: it's enough to use `WithRegisterInfo` to specify a `RegistryInfo` with this field set to true.